### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by MatthewMasarik-NOAA** [NOAA-EMC/WW3#579]
**Describe the bug**
A memory leak in WW3 was detected while running as component of UFS.  This was reported in UFS issue [#779](https://github.com/ufs-community/ufs-weather-model/discussions/779).  

**To Reproduce**
* Run a coupled UFS regression test which has waves enabled.
  * In `nems.configure.cpld_wave.IN`, set WAV section `MemoryProfiling = true`. 
  * Review the profiling output in the PET\* files, in particular the `VmPeak` memory for WW3.  These numbers will increase over the life of the simulation.

**Expected behavior**
Memory use will increase during the simulation.

**Screenshots**
Reference plots in UFS issue [#779](https://github.com/ufs-community/ufs-weather-model/discussions/779).

**Additional context**
N/A.

**Comment by MatthewMasarik-NOAA** [comment]
@DeniseWorthen I'm currently looking at memory use for running WW3 standalone, vs. coupled with the current wmesmfmd cap.  This has me curious how the vmsize behaves with your new mesh cap.  Have you by any chance looked at the memory profile for WW3 w/ the new cap?  I believe you said there are no RT's that support the new cap currently.. if there is a canned case I could run I'd be happy to do that and look at the memory use.

**Comment by DeniseWorthen** [comment]
@MatthewMasarik-NOAA No, I have not done any profiling with the mesh cap. There are no standalone RTs using the mesh cap. It is only being used in the coupled model and there are RTs there (eg cpld_control_p8). 

Setting the mesh cap to run a standalone RT will be needed I assume to merge into develop, so perhaps that should also be added as an issue under the project now that I think of it.

**Comment by MatthewMasarik-NOAA** [comment]
@DeniseWorthen Oh, my mistake.. I misunderstood.  I thought you meant it did not run under any coupled model RT's yet.  That's great.  I'll profile it with the coupled RT's.  Thanks! 👍

**Comment by mickaelaccensi** [comment]
@MatthewMasarik-NOAA is this issue solved ?

**Comment by MatthewMasarik-NOAA** [comment]
@mickaelaccensi, this has not been completely resolved and should stay open.  It's thought that if a leak does exist it's relatively small and not of significant concern.  The priority has been lowered on this task accordingly, but it's still open, and I hope to return to it after some of our operational deadlines are met.
